fmriprep/cli/parser.py

```diff
@@ -0,0 +1,405 @@
+(reason: %s).
+That means some severe flaw was found in it and we strongly
+discourage its usage.""" % (config.environment.version, _reason), file=sys.stderr)
+
+    return parser
+
+
+def parse_args(args=None, namespace=None):
+    """Parse args and run further checks on the command line."""
+    import logging
+    parser = _build_parser()
+    opts = parser.parse_args(args, namespace)
+    config.execution.log_level = int(max(25 - 5 * opts.verbose_count, logging.DEBUG))
+    config.from_dict(vars(opts))
+
+    config.init_loggers()
+
+    # Retrieve logging level
+    build_log = config.loggers.cli
+
+    if config.execution.fs_license_file is None:
+        raise RuntimeError("""\
+ERROR: a valid license file is required for FreeSurfer to run. fMRIPrep looked for an existing \
+license file at several paths, in this order: 1) command line argument ``--fs-license-file``; \
+2) ``$FS_LICENSE`` environment variable; and 3) the ``$FREESURFER_HOME/license.txt`` path. Get it \
+(for free) by registering at https://surfer.nmr.mgh.harvard.edu/registration.html""")
+    os.environ['FS_LICENSE'] = str(config.execution.fs_license_file)
+
+    # Load base plugin_settings from file if --use-plugin
+    if opts.use_plugin is not None:
+        from yaml import load as loadyml
+        with open(opts.use_plugin) as f:
+            plugin_settings = loadyml(f)
+        _plugin = plugin_settings.get('plugin')
+        if _plugin:
+            config.nipype.plugin = _plugin
+            config.nipype.plugin_args = plugin_settings.get('plugin_args', {})
+            config.nipype.nprocs = config.nipype.plugin_args.get('nprocs', config.nipype.nprocs)
+
+    # Resource management options
+    # Note that we're making strong assumptions about valid plugin args
+    # This may need to be revisited if people try to use batch plugins
+    if 1 < config.nipype.nprocs < config.nipype.omp_nthreads:
+        build_log.warning(
+            'Per-process threads (--omp-nthreads=%d) exceed total '
+            'threads (--nthreads/--n_cpus=%d)', config.nipype.omp_nthread, config.nipype.nprocs)
+
+    bids_dir = config.execution.bids_dir
+    output_dir = config.execution.output_dir
+    work_dir = config.execution.work_dir
+    version = config.environment.version
+
+    if config.execution.fs_subjects_dir is None:
+        config.execution.fs_subjects_dir = output_dir / 'freesurfer'
+
+    # Wipe out existing work_dir
+    if opts.clean_workdir and work_dir.exists():
+        from niworkflows.utils.misc import clean_directory
+        build_log.log("Clearing previous fMRIPrep working directory: %s",
+                      work_dir)
+        if not clean_directory(work_dir):
+            build_log.warning("Could not clear all contents of working directory: %s",
+                              work_dir)
+
+    # Ensure input and output folders are not the same
+    if output_dir == bids_dir:
+        config.loggers.cli.error(
```

Comment:
I mistakenly changed these, they should raise `parser.error`s

```suggestion
        raise parser.error(
```